ci/cloudbuild/dockerfiles/fedora-37-bazel.Dockerfile

```diff
@@ -51,3 +51,14 @@ ENV PATH=${CLOUD_SDK_LOCATION}/bin:${PATH}
 RUN curl -o /usr/bin/bazelisk -sSL "https://github.com/bazelbuild/bazelisk/releases/download/v1.17.0/bazelisk-linux-${ARCH}" && \
     chmod +x /usr/bin/bazelisk && \
     ln -s /usr/bin/bazelisk /usr/bin/bazel
+
+# Download the packages needed to run Bigtable conformance tests.
+WORKDIR /var/tmp/downloads
+RUN wget -O go.tgz https://go.dev/dl/go1.20.5.linux-amd64.tar.gz
+RUN tar -C /usr/local/ -xzf go.tgz
+ENV GO_LOCATION=/usr/local/go
+ENV PATH=${GO_LOCATION}/bin:${PATH}
+RUN go version
+WORKDIR /var/tmp/downloads/cloud-bigtable-clients-test
+RUN curl -fsSL https://github.com/googleapis/cloud-bigtable-clients-test/archive/v0.0.1.tar.gz | \
+    tar -xzf - --strip-components=1
```

Comment:
Something here appears to leave a bunch of unwritable directories in `build-out/${DISTRO_FLAG}-${BUILD_FLAG}/h/go/pkg/mod`

Reply:
Where did you see that, and what do the unwritable directories look like?

Reply:
Hopefully this gives you enough information to go on.
```
$ ci/cloudbuild/build.sh --docker --trigger=asan-pr
    : 
ok      github.com/googleapis/cloud-bigtable-clients-test/tests 34.414s
==> 🕑 asan completed in 149.316 seconds
$ find build-out/fedora-37-bazel-asan/ -type d \! -perm -u=w  
build-out/fedora-37-bazel-asan/h/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1
build-out/fedora-37-bazel-asan/h/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/.github
    : 
build-out/fedora-37-bazel-asan/h/go/pkg/mod/github.com/golang/protobuf@v1.5.3/protoc-gen-go/generator/internal/remap
build-out/fedora-37-bazel-asan/h/go/pkg/mod/github.com/golang/protobuf@v1.5.3/protoc-gen-go/grpc
$ ls -ld build-out/fedora-37-bazel-asan/h/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1
dr-xr-xr-x 3 bww eng 4096 Jun 28 12:46 build-out/fedora-37-bazel-asan/h/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1
$ rm -fr build-out/
rm: cannot remove 'build-out/fedora-37-bazel-asan/h/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/writerc.go': Permission denied
rm: cannot remove 'build-out/fedora-37-bazel-asan/h/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/README.md': Permission denied
    : 
    : 
```
There are 1354 unwriteable directories in that list.